File: 프로그래머스/lv0/120860. 직사각형 넓이 구하기/직사각형 넓이 구하기.java

class Solution {
    public int solution(int[][] dots) {
        int answer = 0;
        int w = 0;
        int h = 0;
        int x = dots[0][0];
        int y = dots[0][1];
        
        for(int i=0;i<dots.length;i++){
            if(dots[i][0] != x) w = Math.abs(x-dots[i][0]);
            if(dots[i][1] != y) h = Math.abs(y-dots[i][1]); 
        }
        
        answer = w * h;
        
        return answer;
    }
}

File: 프로그래머스/0/120860. 직사각형 넓이 구하기/직사각형 넓이 구하기.js
function solution(dots) {
    var answer = 1;
    let map1 = new Map();
    let map2 = new Map();
    let lengthX = 0;
    let lengthY = 0;
    
    dots.forEach(item => {
        if(map1.has(item[1])){
            lengthX = Math.abs(map1.get(item[1]) - item[0]);
        }else{
            map1.set(item[1],item[0])
        }
        
        if(map2.has(item[0])){
            lengthY = Math.abs(map2.get(item[0]) - item[1]);
        }else{
            map2.set(item[0],item[1])
        }
    })
    
    return answer*lengthX*lengthY;
}